src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -608,7 +608,57 @@ private void CheckNewModifier(Symbol symbol, bool isNew, DiagnosticBag diagnosti
 
                     if ((object)associatedPropertyOrEvent == null)
                     {
-                        diagnostics.Add(ErrorCode.ERR_OverrideNotExpected, overridingMemberLocation, overridingMember);
+                        bool suppressError = false;
+                        if (overridingMemberIsMethod)
+                        {
+                            foreach (var parameterType in ((MethodSymbol) overridingMember).ParameterTypes)
+                            {
+                                if (IsOrContainsErrorType(parameterType.TypeSymbol))
+                                {
+                                    suppressError = true; // The parameter type must be fixed before the override can be found, so suppress error
+                                    break;
+                                }
+                            }
+                            
+                        }
+                        if(!suppressError)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_OverrideNotExpected, overridingMemberLocation, overridingMember);
+                        }
+
+                        bool IsOrContainsErrorType(TypeSymbol symbol)
```

Comment:
The local function should be moved to the end of the method and the first letter lower-cased, by our conventions. Thanks

Reply:
The purpose of this function is to visit all the nested types. We have a method for that already: `VisitType`.
It should be possible to use it instead.